flux-bin/src/utils.rs

```diff
@@ -0,0 +1,108 @@
+use std::{env, error::Error, ffi::OsString, fs, path::PathBuf};
+
+#[cfg(target_os = "windows")]
+pub const LIB_PATH: &str = "PATH";
+#[cfg(target_os = "linux")]
+pub const LIB_PATH: &str = "LD_LIBRARY_PATH";
+#[cfg(target_os = "macos")]
+pub const LIB_PATH: &str = "DYLD_FALLBACK_LIBRARY_PATH";
+
+pub const EXIT_ERR: i32 = -1;
+
+// It's better than using .expect(), right?
+pub fn report_err(message: &str, err: impl Error) -> i32 {
```

Comment:
Maybe better to just return `Result<T, Box<dyn std::error::Error>>` everywhere and then print at the top level. The `description` will probably be good enough (and maybe better than our custom messages)  so no point in categorizing all the errors.

Reply:
Yeah I think that makes sense. I'm rendering the `impl Error` in addition to outputting a message, but looking through the file, it seems like there isn't much helpful.

Reply:
We now use `anyhow` to collect the errors.